_posts/2025-05-20-highlights-otp-28.md

```diff
@@ -0,0 +1,662 @@
+
+* [Priority Messages](#priority-messages)
+* [Improvements of Comprehensions](#improvements-of-comprehensions)
+* [Smarter Error Suggestions](#smarter-error-suggestions)
+* [Improvements to the Shell](#improvements-to-the-shell)
+* [New `hibernate/0`](#new-hibernate0)
+* [Warnings for Use of Old-style Catch](#warnings-for-use-of-old-style-catch)
+* [PCRE2](#pcre2)
+* [Based Floating Point Literals](#based-floating-point-literals)
+* [Nominal Types](#nominal-types)
+
+
+# Priority Messages
+
+Sometimes, it is important for urgent messages to _skip the queue_ and
+be read by the receiving process as soon as possible. Erlang/OTP 28 introduces
+priority messages, an opt-in mechanism that allows the receiving process
+to let certain messages get priority status.
+
+By default, all messages are inserted to the end of the message queue of
+a process. This can become cumbersome when the queue is long. An urgent
+message may need to be read as soon as possible.
+
+Priority message solves this problem by letting selected messages be inserted
+before all ordinary messages, but still in the order they are received.
+
+A receiver process can allow other processes to send priority message
+to itself in two simple steps. The first step is to create a process alias
+using `alias/1`:
+
+````
+PrioAlias = alias([priority])
+````
+
+This alias can then be distributed to other processes that should be able
+to send priority messages to the receiver process. A sender process can
+send a priority message by using `erlang:send/3`, passing the `PrioAlias`
+as the first argument, and the option `priority` in the option list as
+the third argument:
+
+````
+erlang:send(PrioAlias, Message, [priority])
+````
+
+In this way, messages sent to a priority alias with the `priority` flag will
+be inserted before ordinary messages in the message queue. Other processes
+can still send ordinary messages to the priority alias by not using the
+`priority` flag. If a message is sent to the priority alias without using
+the `priority` flag, it will be treated as an ordinary message.
+
+It is also possible to send an exit or monitor signal as a priority signal,
+like this:
+
+````
+exit(PrioAlias, Message, [priority])
+````
+
+If the receiver process wants to stop receiving priority messages, it
+can do so by deactivating its priority alias:
+
+````
+_ = unalias(PrioAlias)
+````
+
+After this, no priority message can be sent to the receiver process, because
+the priority alias is no longer active. The receiver process can activate
+and deactivate its priority alias again at any time.
+
+Priority messages respect Erlang's existing guarantee: Signals still arrive
+in the same order as they are sent, if they arrive at all. This change
+only affects where messages are inserted in the queue. Performance-wise,
+this feature preserves Erlang's selective receive optimization. There is
+no performance penalty for ordinary messages or priority messages.
+
+
+# Improvements of Comprehensions
```

Comment:
Maybe link to https://www.erlang.org/doc/system/expressions.html#comprehensions somewhere below?